crossplane/ext/lua.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+
+from .abstract import ExternalDirective
+from crossplane import lexer, builder, objects
+from crossplane.errors import NgxParserBaseException
+from crossplane.ext.abstract import CrossplaneExtension
+
+
+class LuaBlockPlugin(CrossplaneExtension):
+    """
+    This plugin adds special handling for Lua code block directives (*_by_lua_block)
+    We don't need to handle non-block or file directives because those are parsed
+    correctly by base Crossplane functionality.
+    """
+
+    def __init__(self):
+        super(LuaBlockPlugin, self).__init__()
+
+        # todo maybe: populate the actual directive bit masks if analyzer/parser logic is needed
+        self.directives = {
+            'access_by_lua_block' : [],
+            'balancer_by_lua_block' : [],
+            'body_filter_by_lua_block' : [],
+            'content_by_lua_block' : [],
+            'header_filter_by_lua_block' : [],
+            'init_by_lua_block' : [],
+            'init_worker_by_lua_block' : [],
+            'log_by_lua_block' : [],
+            'rewrite_by_lua_block' : [],
+            'set_by_lua_block' : [],
+            'ssl_certificate_by_lua_block' : [],
+            'ssl_session_fetch_by_lua_block' : [],
+            'ssl_session_store_by_lua_block' : [],
+        }
```

Comment:
It'd probably make more sense to make this `self.directives` thing into a class variable instead of an instance variable